qlight/config.go

```diff
@@ -0,0 +1,101 @@
+package qlight
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/plugin/security"
+)
+
+type TLSConfig struct {
+	CACertFileName       string
+	ClientCACertFileName string
+	ClientAuth           int
+	CertFileName         string
+	KeyFileName          string
+	InsecureSkipVerify   bool
+	CipherSuites         string
+	ServerName           string
+}
+
+func ConfigureQLightTLSConfig(config *TLSConfig) (*tls.Config, error) {
+	if config.InsecureSkipVerify {
+		return &tls.Config{
+			InsecureSkipVerify: true,
+		}, nil
+	}
+	var (
+		CA_Pool *x509.CertPool
+		err     error
+	)
+	if len(config.CACertFileName) > 0 {
+		CA_Pool, err = x509.SystemCertPool()
+		if err != nil {
+			CA_Pool = x509.NewCertPool()
+		}
+		cert, err := ioutil.ReadFile(config.CACertFileName)
+		if err != nil {
+			return nil, err
+		}
+		CA_Pool.AppendCertsFromPEM(cert)
+	}
+
+	var (
+		ClientCA_Pool *x509.CertPool
+		ClientAuth    tls.ClientAuthType
+	)
+	if len(config.ClientCACertFileName) > 0 {
+		ClientCA_Pool, err = x509.SystemCertPool()
+		if err != nil {
+			ClientCA_Pool = x509.NewCertPool()
+		}
+		cert, err := ioutil.ReadFile(config.ClientCACertFileName)
```

Comment:
in go 1.16, it simply calls os.ReadFile, so we could use that directly now.
```suggestion
		cert, err := os.ReadFile(config.ClientCACertFileName)
```

Reply:
suggestion applied